src/statement.d

```diff
@@ -3226,12 +3227,15 @@ public:
              * result of condition.
              */
             match = new VarDeclaration(loc, prm.type, prm.ident, new ExpInitializer(loc, condition));
-            match.parent = sc.func;
+            match.parent = scd.func;
             match.storage_class |= prm.storageClass;
+            match.semantic(scd);
```

Comment:
When `match.semantic(scd)` failed to analysis, match.type will be `TypeError`. By the separation of `match.semantic`, the `VarExp` will correctly get the error type and won't re-run semantic analysis on the invalid `prm.type`.